docs/DeveloperGuide.md

```diff
@@ -1,38 +1,416 @@
 # Developer Guide
+## Contents
+* [Acknowledgements](#acknowledgements)
+* [Design](#design)
+    * [Architecture](#architecture)
+    * [Manager Component](#manager-component)
+    * [Profile Component](#profile-component)
+    * [TextUI Component](#textui-component)
+    * [Storage Component](#storage-component)
+    * [Parser Component](#parser-component)
+    * [Command Component](#command-component)
+* Implementation
```

Comment:
Inconsistent formatting: I think there's a missing a hyperlink to the Implementation section

Reply:
Might want to use numbers instead of bullet points given the comprehensive nature of your guide which makes very hard to follow through